### PR TITLE
Interceptingeventsapps

### PR DESCRIPTION
**Pull request by Saif-Shines**
- Phase 2 level complete
- Working gif added

### CODE REVIEW

File: Freshdesk/v2/intercepting_events_app/app/app.js [hunk 1]
```diff
@@ -1,70 +1,85 @@
...
+    client.events.on("app.activated", function () {
+      client.events.on("ticket.closeTicketClick", timerValidation, {
+        intercept: true
+      });
+      client.events.on("ticket.propertiesUpdated", propertiesUpdatedCallback, {
+        intercept: true
+      });
     });
   });
 });
```
Comment (velmurugan-balasubramanian) [inline Freshdesk/v2/intercepting_events_app/app/app.js]: var to let ?

**Comment by Saif-Shines** [comment]
@velmurugan-balasubramanian 
I did not understand both the comments. Adding some more context could help me.

File: Freshdesk/v2/intercepting_events_app/app/app.js [hunk 2]
```diff
@@ -1,70 +1,85 @@
...
-      // request api to get time_entries for the ticket
-      const baseUrl = `https://<%= iparam.freshdesk_domain %>.freshdesk.com`;
-      const url = `${baseUrl}/api/v2/tickets/${ticketData.ticket.id}/time_entries`;
-      const options = {
-        "headers" : {
+var timerValidation = function timerValidation(event) {
+  client.data.get("ticket").then(
+    function (ticketData) {
+      var baseUrl = "https://<%= iparam.freshdesk_domain %>.freshdesk.com";
+      var url = ""
```
Comment (hemchander23) [inline Freshdesk/v2/intercepting_events_app/app/app.js]: Can we boil this down? baseUrl is already a string. Do we still need `"".concat..`

File: Freshdesk/v2/intercepting_events_app/config/iparam_test_data.json [hunk 3]
```diff
@@ -1,4 +1,4 @@
 {
-  "freshdesk_domain": "Enter your freshdesk domain here",
-  "freshdesk_key": "Please enter your API Key"
-}
\ No newline at end of file
+  "freshdesk_domain": "Please enter domain name",
```
Comment (hemchander23) [inline Freshdesk/v2/intercepting_events_app/config/iparam_test_data.json]: Follow `<ENTER_YOUR_FRESHDESK_DOMAIN_HERE>` style. Current style would confuse beginners on understanding whether it's a filler or a feature.

**Pull request merged**